backend/src/test/java/de/laura/project/controller/TriviaControllerTest.java

```diff
@@ -39,45 +45,104 @@ private String getUrl() {
     private TestRestTemplate testRestTemplate;
 
     @Autowired
-    private TriviaService triviaService;
+    private TempTriviaQuestionDB tempTriviaQuestionDB;
+
+    @BeforeEach
+    public void setup() {
+    tempTriviaQuestionDB.clear();
+    }
 
     @Test
-    @DisplayName("Get to /questions returns list of questions from Trivia API")
+    @DisplayName("Calls question set on the basis of the user's input")
 
-    public void returnQuestionSet() {
+    public void callQuestionSet() {
 
         //GIVEN
-        String url = "https://opentdb.com/api.php?amount=10&category=12&difficulty=medium&type=multiple";
-
-        ArrayList<String> answerForTriviaQuestionSet = new ArrayList<>();
-        answerForTriviaQuestionSet.add("correct answer");
-        answerForTriviaQuestionSet.add("answerOne");
-        answerForTriviaQuestionSet.add("answerTwo");
-        answerForTriviaQuestionSet.add("answerThree");
-
-        ArrayList<String> anotherAnswerForTriviaQuestionSet = new ArrayList<>();
-        answerForTriviaQuestionSet.add("another correct answer");
-        answerForTriviaQuestionSet.add("answerOne");
-        answerForTriviaQuestionSet.add("answerTwo");
-        answerForTriviaQuestionSet.add("answerThree");
+        String ApiUrl = "https://opentdb.com/api.php?amount=10&category=12&difficulty=medium&type=multiple";
 
         List<String> answersForApiDataList = List.of("answerOne", "answerTwo", "answerThree");
 
-        List<TriviaApiData> mockedTriviaApiDataList = List.of(
-                new TriviaApiData("12", "multiple choice", "easy", "This is an awesome question", "correct answer", answersForApiDataList),
-                new TriviaApiData("12", "multiple choice", "easy", "This is another awesome question", "another correct answer", answersForApiDataList));
+        ArrayList<TriviaApiData> mockedTriviaApiDataList = new ArrayList<>();
```

Comment:
```suggestion
        ArrayList<TriviaApiData> mockedTriviaApiDataList = new ArrayList<>(List.of(
             new TriviaApiData("12", "multiple choice", "easy", "This is an awesome question", "correct answer", answersForApiDataList)
        ));
```

you can also initialise your ArrayList inline making it a bit easier to read